projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -148,8 +149,10 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
     @Input(`igxDateTimeEditor`)
     public set inputFormat(value: string) {
         if (value) {
-            this.setMask(value);
-            this._inputFormat = value;
+            this._userSetFormat = value;
+            this._inputFormat = DateTimeUtil.getNumericInputFormat(this.locale, value);
```

Comment:
Add to the list with user set and changing the actual input format instead of the default and supporting non-numeric formats and so on :)